File: typescript/src/models/refund.ts
// this file is @generated
import { type RefundType, RefundTypeSerializer } from "./refundType";

export interface Refund {
  type: RefundType;
  amount: number;
}

export const RefundSerializer = {
  _fromJsonObject(object: any): Refund {
    return {
      type: RefundTypeSerializer._fromJsonObject(object["type"]),
      amount: object["amount"],
    };
  },

  _toJsonObject(self: Refund): any {
    return {
      type: RefundTypeSerializer._toJsonObject(self.type),
      amount: self.amount,
    };
  },
};

File: rust/src/models/refund.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::refund_type::RefundType;

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct Refund {
    pub r#type: RefundType,
    pub amount: i32,
}
